bundle/markov/metrics/s3_metrics.py

```diff
@@ -27,20 +44,26 @@
 from markov.gazebo_tracker.abs_tracker import AbstractTracker
 from markov.gazebo_tracker.constants import TrackerPriority
 from markov.track_geom.track_data import TrackData
-from markov.boto.s3.constants import (SIMTRACE_EVAL_LOCAL_PATH_FORMAT,
-                                      SIMTRACE_TRAINING_LOCAL_PATH_FORMAT)
+from markov.boto.s3.constants import (SIMTRACE_EPISODE_EVAL_LOCAL_PATH_FORMAT, SIMTRACE_EPISODE_TRAINING_LOCAL_PATH_FORMAT, 
+                                      SIMTRACE_EVAL_LOCAL_PATH_FORMAT, SIMTRACE_TRAINING_LOCAL_PATH_FORMAT)
+from markov.boto.firehose.files.metrics import FirehoseMetrics
+from markov.boto.firehose.files.simtrace import FirehoseSimtrace
 from markov.boto.s3.files.metrics import Metrics
 from telegraf.client import TelegrafClient    
 LOGGER = Logger(__name__, logging.INFO).get_logger()
 
+enable_episode_simtrace = str2bool(rospy.get_param("ENABLE_EPISODE_SIMTRACE", False)) # feature flag
+enable_firehose_upload = str2bool(rospy.get_param("ENABLE_FIREHOSE_UPLOAD", False)) # feature flag
+
+
 #! TODO this needs to be removed after muti part is fixed, note we don't have
 # agent name here, but we can add it to the step metrics if needed
 def sim_trace_log(sim_trace_dict):
     '''Logs the step metrics to cloud watch
        sim_trace_dict - Ordered dict containing the step metrics, note order must match
                         precision in the string
     '''
-    LOGGER.info('SIM_TRACE_LOG:%d,%d,%.4f,%.4f,%.4f,%.2f,%.2f,%s,%.4f,%s,%s,%.4f,%d,%.2f,%s,%s,%.2f\n' % \
+    LOGGER.info('SIM_TRACE_LOG:%d,%d,%.4f,%.4f,%.4f,%.2f,%.2f,%s,%.4f,%s,%s,%.4f,%d,%.2f,%s,%s,%.2f,%d\n' % \
```

Comment:
What does this actually do?

Reply:
Field covers the number of object resets. Sim-trace can still be loaded, hence no issue.